x-pack/plugins/maps/common/content_management/types.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type {
+  SavedObject,
+  SavedObjectReference,
+  SavedObjectsResolveResponse,
+  SavedObjectsUpdateResponse,
+} from '@kbn/core-saved-objects-api-server';
+import type { SavedObjectsMigrationVersion } from '@kbn/core-saved-objects-common';
+
+import type {
+  GetIn,
+  CreateIn,
+  SearchIn,
+  UpdateIn,
+  DeleteIn,
+} from '@kbn/content-management-plugin/common';
+
+import { MapSavedObjectAttributes } from '../map_saved_object_type';
+
+interface Reference {
+  type: string;
+  id: string;
+}
+
+export type MapSavedObject = SavedObject<MapSavedObjectAttributes>;
+
+export type MapContentType = 'map';
+
+// ----------- GET --------------
+
+export type MapGetIn = GetIn<MapContentType>;
```

Comment:
A lot of these types could be abstracted. The SO specific items are rather small.

Reply:
None of those type are required. There are there to be explicit and make our lives easier. You can use the plain `GetIn` and other generics exposed from the `content_management` plugin if you prefer. I think it is nice (and better) to have dedicated types for each content. If anything changes in the interfaces, there is only a single place to look at. It is also better for full service versioning that I am going to add.